### PR TITLE
Yubikey sample

### PR DESCRIPTION
**Pull request by yschimke**
Sample for using hardware tokens on JVM to achieve client auth.

### CODE REVIEW

**Comment by yschimke** [comment]
Output

```

<br><span class="sslsuccess">SSL Authentication OK!</span><br><br>Technical information follows :<pre>Array
(
    [modHeadersAvailable] => true
    [HTTPS] => on
    [SSL_TLS_SNI] => prod.idrix.eu
    [SSL_SERVER_S_DN_OU] => Domain Control Validated
    [SSL_SERVER_S_DN_OU_1] => PositiveSSL
    [SSL_SERVER_S_DN_CN] => prod.idrix.eu
    [SSL_SERVER_I_DN_C] => GB
    [SSL_SERVER_I_DN_ST] => Greater Manchester
    [SSL_SERVER_I_DN_L] => Salford
    [SSL_SERVER_I_DN_O] => COMODO CA Limited
    [SSL_SERVER_I_DN_CN] => COMODO RSA Domain Validation Secure Server CA
    [SSL_CLIENT_S_DN_CN] => Yuri Schimke
```

**Comment by yschimke** [comment]
Works with PIV compatible devices https://www.yubico.com/products/compare-yubikey-5-series/

File: samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import javax.security.auth.callback.Callback
+import javax.security.auth.callback.CallbackHandler
+import javax.security.auth.callback.PasswordCallback
+import javax.security.auth.callback.UnsupportedCallbackException
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.internal.SuppressSignatureCheck
+import okhttp3.internal.platform.Platform
+
+object ConsoleCallbackHandler : CallbackHandler {
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt:36]: Seems like this should be defined after `YubikeyClientAuth`, the class that names the file & has documentation?

File: samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    // May fail with ProviderException with root cause like
+    // sun.security.pkcs11.wrapper.PKCS11Exception: CKR_SLOT_ID_INVALID
+    val pkcs11 = Security.getProvider("SunPKCS11").configure(config)
+    Security.addProvider(pkcs11)
+
+    val callbackHandler = ConsoleCallbackHandler
+
+    val builderList: List<KeyStore.Builder> = Arrays.asList(
+        KeyStore.Builder.newInstance("PKCS11", null, KeyStore.CallbackHandlerProtection(callbackHandler))
+//        KeyStore.Builder.newInstance("PKCS12", null, File("keystore.p12"), PasswordProtection("rosebud".toCharArray()))
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt:85]: remove or explain?

File: samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    )
+
+    val kmf = KeyManagerFactory.getInstance("NewSunX509")
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt:88]: big names please? keyManagerFactory or factory

File: samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    val sslContext = SSLContext.getInstance("TLS")
+    sslContext.init(arrayOf(km), arrayOf(tm), SecureRandom())
+
+    val client = OkHttpClient.Builder()
+        .sslSocketFactory(sslContext.socketFactory, tm)
+        .build()
+
+    val request = Request.Builder()
+        .url("https://prod.idrix.eu/secure/")
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt:102]: Is this a reasonable endpoint for developers to play with?

File: samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  }
+}
+
+/**
+ * Example of using a hardware key to perform client auth.
+ * Prefer recent JDK builds, and results are temperamental to slight environment changes.
+ * Different instructions and configuration may be required for other hardware devices.
+ *
+ * Using a yubikey device as a SSL key store.
+ * https://lauri.võsandi.com/2017/03/yubikey-for-ssh-auth.html
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt:61]: neat domain name

File: samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt (hunk 4 above)
Comment (yschimke) [reply samples/guide/src/main/java/okhttp3/recipes/kt/YubikeyClientAuth.kt:102]: It's specifically a test URL.

**Pull request merged**